lib/feedx/compression.rb

```diff
@@ -5,22 +5,29 @@ module Compression
     autoload :Gzip, 'feedx/compression/gzip'
 
     class << self
+      def validate!(kind)
+        raise ArgumentError, "#{kind} does not implement #reader(io, &block)" unless kind.respond_to?(:reader)
+        raise ArgumentError, "#{kind} does not implement #writer(io, &block)" unless kind.respond_to?(:writer)
+
+        kind
+      end
```

Comment:
`Compression.validate!` doesn't seem to be used anywhere?
There's no registry - no `Compression.register` (like `Format.register`) where this is meant to be called.

UPD: found where it's called. But still - don't you want a `Compression` registry? (it really looks like a forgotten thing to implement)

Reply:
not forgotten, just left it until we actually need it tbh